### PR TITLE
Feature/pbspf review and feedback system

### PR DESCRIPTION
**Pull request by newtronahmed**
This pull request includes several significant changes across multiple files to improve configuration, dependency management, and functionality for logging, API responses, and review management.
### API Enhancements:
* Enhanced `ProviderServiceController` by adding detailed API documentation using Swagger annotations and refactoring methods for better response handling. [[1]](diffhunk://#diff-f398bfa61ab279d9cbe516af3e15584a0d0830029a15ad8a3aa91f762bc02622R3-R21) [[2]](diffhunk://#diff-f398bfa61ab279d9cbe516af3e15584a0d0830029a15ad8a3aa91f762bc02622R37-R143)
* Added a new `ReviewController` to manage reviews, including endpoints for creating reviews, fetching reviews, and performing sentiment analysis.

### Dependency Management:
* Added Spring Cloud dependencies and other necessary dependencies like `stanford-corenlp` and `spring-boot-starter-actuator` in `management/pom.xml`. [[1]](diffhunk://#diff-0b477b92e5f429d44544f1c64fbf3bb21ed34ab241afa28ac645b9661e6685b7R31-R43) [[2]](diffhunk://#diff-0b477b92e5f429d44544f1c64fbf3bb21ed34ab241afa28ac645b9661e6685b7R104-R114) [[3]](diffhunk://#diff-0b477b92e5f429d44544f1c64fbf3bb21ed34ab241afa28ac645b9661e6685b7R126-R139)

### Logging and Interceptors:
* Introduced `RequestLoggingInterceptor` to log incoming requests and their headers.
* Updated `WebConfig` to register `RequestLoggingInterceptor` and added a bean for `HttpExchangeRepository`.

### CODE REVIEW

File: authservice/src/main/resources/application.yml
Comment (Zena4L) [inline authservice/src/main/resources/application.yml:1]: to make it simple to change profiles, use environment variables instead.

File: management/src/main/java/team/proximity/management/config/RequestLoggingInterceptor.java [hunk 1]
```diff
@@ -23,8 +23,8 @@ public boolean preHandle(HttpServletRequest request, HttpServletResponse respons
         request.getHeaderNames().asIterator().forEachRemaining(headerName ->
                 logger.info("Header: {} = {}", headerName, request.getHeader(headerName))
         );
-        String requestBody = request.getReader().lines().collect(Collectors.joining(System.lineSeparator()));
-        logger.info("Request Body: {}", requestBody);
+//        String requestBody = request.getReader().lines().collect(Collectors.joining(System.lineSeparator()));
```
Comment (Zena4L) [inline management/src/main/java/team/proximity/management/config/RequestLoggingInterceptor.java:26]: Clean out comments if not needed

File: management/src/main/java/team/proximity/management/controllers/ReviewController.java
Comment (Zena4L) [inline management/src/main/java/team/proximity/management/controllers/ReviewController.java:1]: clean out comments

File: management/src/main/java/team/proximity/management/services/ReviewService.java
Comment (Zena4L) [inline management/src/main/java/team/proximity/management/services/ReviewService.java:1]: clean up commented codes

**Review by Zena4L — CHANGES_REQUESTED** [review]
Few comments

File: authservice/src/main/resources/application.yml
Comment (newtronahmed) [reply authservice/src/main/resources/application.yml:1]: Yeah sure

**Pull request merged**